app/feature/feature-travel-certificate/src/main/kotlin/com/hedvig/android/feature/travelcertificate/ui/choose/ChooseContractForCertificate.kt

```diff
@@ -0,0 +1,156 @@
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.heightIn
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.hedvig.android.core.designsystem.component.button.HedvigContainedButton
+import com.hedvig.android.core.designsystem.component.card.HedvigCard
+import com.hedvig.android.core.designsystem.component.error.HedvigErrorSection
+import com.hedvig.android.core.designsystem.preview.HedvigPreview
+import com.hedvig.android.core.designsystem.theme.HedvigTheme
+import com.hedvig.android.core.ui.SelectIndicationCircle
+import com.hedvig.android.core.ui.clearFocusOnTap
+import com.hedvig.android.core.ui.scaffold.HedvigScaffold
+import com.hedvig.android.data.travelcertificate.ContractEligibleWithAddress
+import com.hedvig.android.feature.travelcertificate.ui.FullScreenLoading
+import hedvig.resources.R
+
+@Composable
+internal fun ChooseContractForCertificateDestination(
+  viewModel: ChooseContractForCertificateViewModel,
+  navigateBack: () -> Unit,
+  onContinue: (String) -> Unit,
+) {
+  val uiState by viewModel.uiState.collectAsStateWithLifecycle()
+  ChooseContractForCertificate(
+    uiState = uiState,
+    navigateBack = navigateBack,
+    onContinue = onContinue,
+    reload = { viewModel.emit(ChooseContractEvent.RetryLoadData) },
+  )
+}
+
+@Composable
+private fun ChooseContractForCertificate(
+  uiState: ChooseContractUiState,
+  navigateBack: () -> Unit,
+  onContinue: (String) -> Unit,
+  reload: () -> Unit,
+) {
+  var selectedContractId by remember {
+    mutableStateOf<String?>(null)
+  }
+  when (uiState) {
+    ChooseContractUiState.Failure -> {
+      ShowFailure(
+        navigateBack = navigateBack,
+        reload = reload,
+      )
+    }
+    ChooseContractUiState.Loading -> {
+      FullScreenLoading()
+    }
+    is ChooseContractUiState.Success -> {
+      HedvigScaffold(
+        navigateUp = navigateBack,
+        modifier = Modifier.clearFocusOnTap(),
+      ) {
+        Spacer(modifier = Modifier.weight(1f))
+        Text(
+          text = stringResource(id = R.string.travel_certificate_select_contract_title),
+          style = MaterialTheme.typography.headlineSmall,
```

Comment:
This one looks like headlineMedium if I read Figma correctly
```suggestion
          style = MaterialTheme.typography.headlineMedium,
```